src/main/java/org/datadog/jenkins/plugins/datadog/model/DatadogLinkAction.java

```diff
@@ -0,0 +1,67 @@
+package org.datadog.jenkins.plugins.datadog.model;
+
+import com.thoughtworks.xstream.XStream;
+import com.thoughtworks.xstream.converters.MarshallingContext;
+import com.thoughtworks.xstream.converters.UnmarshallingContext;
+import com.thoughtworks.xstream.io.HierarchicalStreamReader;
+import com.thoughtworks.xstream.io.HierarchicalStreamWriter;
+import hudson.model.Action;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import org.datadog.jenkins.plugins.datadog.util.conversion.DatadogActionConverter;
+import org.datadog.jenkins.plugins.datadog.util.conversion.VersionedConverter;
+
+public class DatadogLinkAction implements Action {
+
+    private final String url;
+
+    public DatadogLinkAction(BuildData buildData) { // ci.pipeline.url%3A"https%3A%2F%2Fgoogle.com"
+        String query = String.format("ci_level:pipeline @ci.pipeline.name:\"%s\" @ci.pipeline.number:%s", buildData.getJobName(), buildData.getBuildNumber(""));
+        String urlEncodedQuery = URLEncoder.encode(query, StandardCharsets.UTF_8);
+        this.url = String.format("https://app.datadoghq.com/ci/pipeline-executions?query=%s", urlEncodedQuery);
```

Comment:
The host will be different depending on the datacenter and/or, if the org has custom subdomains.
E.g. For orgs in Europe, the url is `https://app.datadoghq.eu/.........`, etc.

This is typically resolved by asking customer to provide a `DD_SITE` environment variable (this does not cover custom subdomains, tho). For Jenkins, this does not seem the best approach, as it has its own config page.

Reply:
I know it's the case when using the intakes, but is it the same when logging in to the UI though? I thought for the app the URL is always the same, as in the static docs we have `https://app.datadoghq.com` hardcoded.
I might be wrong though, will try asking someone who knows